src/helpers/openai.ts

```diff
@@ -0,0 +1,45 @@
+import { readSecret } from "./secrets";
+
+const { OpenAIClient, AzureKeyCredential } = require("@azure/openai");
+
+let messages = [
+    {
+        role: "system",
+        content: "Assistant is a large language model trained by OpenAI.",
+    },
+];
+
+export async function askOpenAI(message: string) {
+    const azureOpenAIEndpoint =
+        process.env.AZURE_OPENAI_ENDPOINT ||
+        readSecret(
+            `/etc/secrets/azure-openai-endpoint-${process.env.TF_WORKSPACE}`
+        );
```

Comment:
query: `Is the first condition i.e. getting from env variables is for local development and the reading from etc/.. is for GCP?`

Reply:
Indeed, `process.env.AZURE_OPENAI_ENDPOINT` for us to pass it during local development and `readSecret` function (which uses `fs.readFileSync`) is used to fetch it from the file location path by the Google Cloud Function.

Secrets in google cloud functions are mounted to a file path we specify, like [here](https://github.com/xplorer-io/xplorers-bot-ts/blob/main/terraform/main.tf#L52). Hence the use of `fs.readFileSync` to read from a file location during runtime. More on this [here](https://cloud.google.com/functions/docs/configuring/secrets).